### PR TITLE
Bump Webpack to version 4

### PR DESCRIPTION
**Pull request by rhystmills**
## What does this change?

This PR bumps Webpack 4, in order to reduce the number of vulnerabilities in the application. 

`node-sass` is removed (solving some high risk vulnerabilities), while a number of other webpack-related dependencies are bumped. All our babel dependencies have been modernised.

## Why?
The two critical vulnerabilities reported by Snyk are related to Webpack and`svg-sprite` (which is also bumped in this PR), and `css-loader`, which couldn't be bumped without bumping Webpack.

Some Sass was changed in order to satisfy the newer version of `sass-loader`.
## How to test

Run `tagmanager` locally according to the instructions in the readme:
- Do the four scripts in `package.json` all run as expected? (n.b. the dev server _was_ and _remains_ slow to start)
- Can you deploy to CODE and successfully run the application there? 
- Do the styles seem to be working?

### CODE REVIEW

File: public/build_config/webpack.prod.conf.js [hunk 1]
```diff
@@ -1,55 +1,68 @@
...
+    rules: [
       {
-        test:    /\.js$/,
+        test: /\.js$/,
+        use: [{
+          loader: "babel-loader",
+          options: {
+            presets: ["@babel/preset-env", "@babel/preset-react"]
+          }
+        }],
```
Comment (andrew-nowak) [inline public/build_config/webpack.prod.conf.js:21]: are there enough overlaps in the webpack conf files to create a common config file and use webpack-merge? maybe one for another pr (upgrading to webpack v5?)

File: public/build_config/webpack.devserver.conf.js [hunk 2]
```diff
@@ -31,15 +31,24 @@ module.exports = {
...
             {
-                test:    /\.jsx?$/,
+                test: /\.(js|jsx)$/,
                 exclude: /node_modules/,
-                loaders: ['react-hot-loader', 'babel-loader?presets[]=es2015&presets[]=react&plugins[]=transform-object-assign']
+                use: [{ loader: 'babel-loader' }]
             },
+            { 
+                test: require.resolve('react'), 
+                use: [{ loader: 'expose-loader', options: 'React' }]},
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply public/build_config/webpack.devserver.conf.js:38]: Appears to be redundant on testing, now removed.

File: public/build_config/webpack.devserver.conf.js [hunk 3]
```diff
@@ -31,15 +31,24 @@ module.exports = {
...
-                test:   require.resolve('react'),
-                loader: 'expose-loader?React'
+              test:    /\.js$/,
+              use: [{
+                loader: "babel-loader",
+                options: {
+                  presets: ["@babel/preset-env", "@babel/preset-react"]
+                }
+              }],
+              exclude: /node_modules/,
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply public/build_config/webpack.devserver.conf.js:47]: Turns out the first rule is the redundant one on testing, now removed.

**Review by andrew-nowak — APPROVED** [review]

File: public/style/_icons.scss [hunk 4]
```diff
@@ -6,153 +6,150 @@
...
-.i-cross-white {
+.i-public--images--icons--cross-white {
 	@extend %svg-common;
-	background-position: 98.46153846153847% 43.75%;
+	background-position: 98.46153846153847% 64.0625%;
 	width: 13px;
 	height: 13px;
 }
 
-.i-delete {
```
Comment (davidfurey) [inline public/style/_icons.scss:58]: @rhystmills I think this has broken the delete icon on parent tags, this should appear in the right hand column here:

Screenshot 2022-10-05 at 11 06 00

**Pull request merged**